api/test/search.test.js

```diff
@@ -6,7 +6,7 @@ describe('search', () => {
     await driver.close();
   });
 
-  test('model search to have 50 results per component type if it exceeds the limit 50', async () => {
+  test('model search should have max 50 results per component type', async () => {
```

Comment:
I'm confused here, the limit was 50 for all results and now it's 50 per component type?

Reply:
The search runs in two steps.

1. Full text search that returns ids from all nodes, including `Metabolite` and `CompartmentalizedMetabolite`. In this case, the limit is 50.
2. More data is fetched for each component type. For the case of `Metabolite`, for example if there are 20 ids found, this could lead to more than 50 results, so the limit is further passed here to make sure it does not exceed 50. The problem is that we were fetching additional data for `Metabolite` ids and `CompartmentalizedMetabolite` ids separately before. So the combination of results returned for `Metabolite` ids and `CompartmentalizedMetabolite` ids are then merged back together, which could result in more than 50 results and that is why the test was failing before.

Does that make sense? This was a tricky implementation so let me know if you want me to elaborate on anything.

Reply:
Ah I see, it does make sense now!